docs/Documentation/Software_Tools/spack.md

```diff
@@ -0,0 +1,306 @@
+## Introduction
+
+Spack is an HPC-centric package manager for acquiring, building, and managing HPC applications as well as all their dependencies, down to the compilers themselves. Like frameworks such as Anaconda, it is associated with a repository of both source-code and binary packages. Builds are fully configurable through a DSL at the command line as well as in YAML files. Maintaining many build-time permutations of packages is simple through an automatic and user-transparent hashing mechanism. The Spack system also automatically creates (customizable) environment modulefiles for each built package.
+
+## Installation
+
+Multiple installations of Spack can easily be kept, and each is separate from the others by virtue of the environment variable `SPACK_ROOT`. 
+All package, build, and modulefile content is kept inside the `SPACK_ROOT` path, so working with different package collections is as simple as setting `SPACK_ROOT` to the appropriate location. 
+The only exception to this orthogonality are `YAML` files in `$HOME/.spack/<platform>`.
+Installing a Spack instance is as easy as
+
+`git clone https://github.com/spack/spack.git`
+
+Once the initial Spack instance is set up, it is easy to create new ones from it through
+
+`spack clone <new_path>`
+
+`SPACK_ROOT` will obviously need to point to `<new_path>` in order to be consistent.
+
+Spack environment setup can be done by sourcing `$SPACK_ROOT/share/spack/setup-env.sh`, or by simply adding `$SPACK_ROOT/bin` to your PATH. 
+We have also created a Lua module to execute the setup-env.sh logic, so it is simple to activate one Spack environment or another through different modules. 
+The template modulefiles are available on request. 
+Two things to keep in mind are
+1.	Bash syntax has some incompatibilities with Lua string definitions, and so it is critical to escape certain patterns.
+2.	As Spack versions progress, the `setup-env.sh` script is subject to change, so it may be necessary to create a new Spack module from the newer `setup-env.sh` logic, using the escaping patterns in the existing template modulefiles as a guide.
+
+## Setting Up Compilers
```

Comment:
We may want to add examples about how to use Eagle compiler modules to set compilers for spack.

Reply:
the same commands described in the document could be followed on eagle and would allow the user to use eagle compilers. Let me know if you're thinking of a specific application of space on eagle.

Reply:
What I meant was that on line 39, we should add something like

"
To list the available compilers on a system, e.g., Eagle, run `module avail`.
"